todomvc.js

```diff
@@ -1,102 +1,91 @@
 import "./styles.css";
 
-import React from "react";
 import ReactDOM from "react-dom";
-import route from "can-route";
-import DefineMap from "can-define/map/";
-import Component from "react-view-model/component";
+import React from "react";
+import Todo from "../models/todo";
 import Create from "./components/create";
 import List from "./components/list";
-import Todo from "./models/todo";
+import { connect, ObserveObject } from "ylem";
+import { getAsync } from 'ylem/property-decorators';
+import route from "can-route";
 
-export const ViewModel = DefineMap.extend('AppVM', {
-	filter: "string",
-	get todosPromise() {
-		if(!this.filter) {
+class Store extends ObserveObject {
+    @getAsync
+	get todosList() {
+		if(!route.data.filter) {
 			return Todo.getList({});
 		}
 		else {
-			return Todo.getList({ complete: this.filter === "complete" });
+			return Todo.getList({ filter: {complete: route.data.filter === "complete" }});
 		}
-	},
-	todosList: {
-		get: function(lastSetValue, resolve) {
-			this.todosPromise.then(resolve);
-		}
-	},
+	}
 	get allChecked() {
 		return this.todosList && this.todosList.allComplete;
-	},
+	}
 	set allChecked(newVal) {
 		this.todosList && this.todosList.updateCompleteTo(newVal);
-	}
-});
+    }
+}
 
-export default class App extends Component {
-	render() {
-		return (
-			<section id="todoapp">
-				<header id="header">
-					<h1>todos</h1>
-					<Create />
-				</header>
+const TodoMVC = ({todosList, allChecked}) => {
+    return (
+    <section id="todoapp">
+        <header id="header">
+            <h1>todos</h1>
+            <Create />
+        </header>
+        <section id="main">
+            <input
+                id="toggle-all"
+                type="checkbox"
+                checked={ allChecked ? true : false }
+                disabled={ todosList && todosList.saving.length }
+            />
+            <label htmlFor="toggle-all">Mark all as complete</label>
+        <List todos={ todosList ? [...todosList] : [] }  />   
```

Comment:
why this `[...todosList]`

Reply:
One of the problems with DefineList is that it returns a new DefineList when mapped, which interefers with react's rendering. This is why I exclusively use ObserveObject and why all official docs use it.

There are a couple of ways around it, this being the simplest. You get an array with the same observable instances inside, and this still reads length/etc so you won't miss any updates. Unfortunately they all violate the rule of special knowledge inside the react component.

I have a bit of code that hooks into DefineList and actually solves the problem (I thought I shared this with you @mikedane?), but it can't go in the ylem core (since we don't depend on can-define) and can't go into the can-define core (since it doesn't depend on react). I haven't been too keen on adding it to the docs either, since new users should be using can-observe. Would it be worth creating a package of this code for the projects that are using can-define?

Reply:
Why is DefineList used here instead of observe.Array? Should/would that fix the problem?

Reply:
The model in this repo is still using can-define. Yes using can-observe (ObserveObject and ObserveArray are both exported by ylem) would fix this. Those are compatible with `realtimeRestModel` right?

Reply:
Since we're already using `ObserveObject` I think `ObserveArray` is the best option, I'll test that it works with `realtimeRestModel` and make the change